### DISCUSSION TITLE
2022/float_95样例在本地测试通过，但是在autograder报错list index out of range?

### DISCUSSION THREAD

**Discussion by tiger9tu** [arcsysu/SYsU-lang#105] [Q&A]
我在本地测试时float_95样例可以通过
![image](https://user-images.githubusercontent.com/76516415/236661544-c55eaa01-577f-4a2c-8fc0-4e0d53585bdd.png)

，但是在autograger报错如下：
![1f4f40f552dee588581f2f7dc67b2e4](https://user-images.githubusercontent.com/76516415/236658598-1162911d-a283-4bb7-a83f-a93bbf5556e6.png)

我尝试用lli分别执行generator输出的IR和clang输出的IR，发现二者的输出一致。
![屏幕截图 2023-05-07 125136](https://user-images.githubusercontent.com/76516415/236658637-d7ccee51-1e15-44b8-9e53-c67d5de134d7.png)
所以请问为什么autograder会报错呢？

**Comment by xuantengh** [comment]
在本地测试时是使用的容器环境吗?

**Reply by tiger9tu** [reply]
刚才试了一下，容器里面也有相同的报错（我之前是wsl环境不报错），那我现在能复现错误了，我在本地容器里面debug试试。

**Comment by tiger9tu** [comment]
![image](https://user-images.githubusercontent.com/76516415/237060508-6dfb561a-f9f0-4d95-99b8-bbc1ed262c81.png)

**Reply by tiger9tu** [reply]
是llvm convertToFloat API的问题，他将浮点数直接转换为0了

**Reply by NyakoApocalipse** [reply]
所以最后怎么解决的呢？（好奇）

**Reply by NyakoApocalipse** [reply]
我遇到和你一样的问题，输出和标准一样但是评测机超index，而且没用convertToFloat，似乎也没有转换为0的问题？

**Reply by tiger9tu** [reply]
最后我自己写了类型转换的逻辑，代替了llvm api

**Reply by NyakoApocalipse** [reply]
LOL, 我的在本地跑不行，放评测机上就能够过了，所以后来我都没管了
